external/storm-jdbc/src/main/java/org/apache/storm/jdbc/bolt/JdbcBolt.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.jdbc.bolt;
+import backtype.storm.topology.OutputFieldsDeclarer;
+import backtype.storm.tuple.Tuple;
+import org.apache.storm.jdbc.common.Column;
+import org.apache.storm.jdbc.mapper.JdbcMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Basic bolt for writing to any Database table.
+ * <p/>
+ * Note: Each JdbcBolt defined in a topology is tied to a specific table.
+ */
+public class JdbcBolt extends AbstractJdbcBolt {
+    private static final Logger LOG = LoggerFactory.getLogger(JdbcBolt.class);
+
+    private String tableName;
+    private JdbcMapper jdbcMapper;
+
+    public JdbcBolt withConfigKey(String configKey) {
```

Comment:
Moved back to constructor. I am not sure what you mean when you say we could just pass the config into the Bolt. Hikari has many configuration https://github.com/brettwooldridge/HikariCP. The configKey is so user can provide any subset of those configs. so the configKey is just name of the config key in your submitted storm-conf, the value of this should be a map of hikari config keys to their corresponding values.

I can extract a connectionProvider interface and create a Hikari implementation that users have to construct to initialize the Bolts. This will mean the bolts will not have a config Key, users will be able to use whatever connection pool they feel most comfortable with and the configuration details will live in the connection provider. Do you think that will be better than what we currently have.